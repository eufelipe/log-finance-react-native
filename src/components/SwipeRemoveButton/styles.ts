import styled from 'styled-components/native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

import {Text, TouchableOpacity} from 'styles/layout';
import Colors from 'styles/colors';

export const Container = styled(TouchableOpacity)`
  margin: ${({theme}) => theme.spacing.small}px
    ${({theme}) => theme.spacing.great}px ${({theme}) => theme.spacing.small}px
    0px;
  padding: 0px ${({theme}) => theme.spacing.tall}px;
  background: ${({theme}) => theme.colors.danger};
  border-radius: ${({theme}) => theme.spacing.great}px;
  justify-content: center;
  align-items: center;
  flex-direction: row;
`;
export const Label = styled(Text)`
  color: ${({theme}) => theme.colors.white};
  font-size: ${({theme}) => theme.fontSizes.small};
`;

export const RemoveIcon = styled(MaterialIcons).attrs({
  size: 24,
  color: Colors.white,
  name: 'delete',
})``;
