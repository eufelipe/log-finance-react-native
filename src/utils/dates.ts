import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
