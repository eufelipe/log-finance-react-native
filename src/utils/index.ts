export {default as number} from './number';
export {default as strings} from './strings';
