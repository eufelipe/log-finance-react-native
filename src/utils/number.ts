export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
