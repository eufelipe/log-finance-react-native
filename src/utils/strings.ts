export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
