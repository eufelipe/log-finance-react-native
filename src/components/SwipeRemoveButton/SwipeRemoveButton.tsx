import React from 'react';
import {useTranslation} from 'react-i18next';

import {Container, RemoveIcon, Label} from './styles';

interface Props {
  onPressRemove: () => void;
}

const SwipeRemoveButton = ({onPressRemove}: Props): JSX.Element => {
  const {t} = useTranslation('global');

  return (
    <Container
      testID="remove-item-button"
      onPress={onPressRemove}
      accessible={true}
      accessibilityLabel={t('accessibility-remove-label')}
      accessibilityHint={t('accessibility-remove-hint')}
    >
      <RemoveIcon />
      <Label>{t('remove-label')}</Label>
    </Container>
  );
};

export default SwipeRemoveButton;
