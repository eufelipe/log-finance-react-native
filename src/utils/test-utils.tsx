/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
