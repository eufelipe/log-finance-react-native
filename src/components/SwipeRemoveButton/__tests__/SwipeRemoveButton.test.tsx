/* eslint-disable @typescript-eslint/unbound-method */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import {render} from 'utils/test-utils';
import SwipeRemoveButton from '../SwipeRemoveButton';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: any) => {
      if (key === 'remove-label') {
        return 'Remover';
      }
      return key;
    },
  }),
}));

const onPressRemove = jest.fn();

describe('SwipeRemoveButton', () => {
  it('Test match snapshot SwipeRemoveButton', () => {
    // given
    const props = {
      onPressRemove,
    };

    // when
    const {toJSON} = render(<SwipeRemoveButton {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test if the description renders correctly', () => {
    // given
    const props = {
      onPressRemove,
    };

    // when
    const {getByText} = render(<SwipeRemoveButton {...props} />);

    // then
    expect(getByText('Remover')).toBeTruthy();
    expect(() => getByText('no exist')).toThrow('Unable to find an element');
  });

  it('Tests if pressing the button is working correctly', () => {
    // given
    const props = {
      onPressRemove,
    };

    // when
    const {getByTestId} = render(<SwipeRemoveButton {...props} />);
    const pressButton = getByTestId('remove-item-button');
    fireEvent.press(pressButton);

    // then
    expect(pressButton).toBeTruthy();
    expect(onPressRemove).toHaveBeenCalled();
    expect(onPressRemove).toHaveBeenCalledTimes(1);
  });
});
