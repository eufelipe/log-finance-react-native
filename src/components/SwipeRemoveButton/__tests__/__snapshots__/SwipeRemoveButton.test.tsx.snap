// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeRemoveButton Test match snapshot SwipeRemoveButton 1`] = `
<View
  accessibilityHint="accessibility-remove-hint"
  accessibilityLabel="accessibility-remove-label"
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E96245",
      "borderRadius": 21.428571428571427,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10.714285714285714,
      "marginLeft": 0,
      "marginRight": 21.428571428571427,
      "marginTop": 10.714285714285714,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 42.857142857142854,
      "paddingRight": 42.857142857142854,
      "paddingTop": 0,
    }
  }
  testID="remove-item-button"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 24,
        },
        Array [
          Object {},
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î¡²
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Poppins-Regular",
          "fontSize": 28,
          "fontWeight": "400",
        },
      ]
    }
  >
    Remover
  </Text>
</View>
`;
